### DISCUSSION TITLE
How to convert output heatmap into keypoint coordinates and confidence scores

### DISCUSSION THREAD

**Discussion by ggggg111** [open-mmlab/mmpose#2758] [Q&A] [answered]
Hi, I am working on a Libtorch-based C++ application using the HRNet32 from torchscript. My question is, how does MMPose convert the heatmap [K, H W] into the keypoints and confidences?

Thank you.

**Comment by Ben-Louis** [comment, marked as answer]
Hi, you can refer to [get_heatmap_maximum](https://github.com/open-mmlab/mmpose/blob/6d10b2ec81da7e252016b3154c7fdb46c403ecd8/mmpose/codecs/utils/post_processing.py#L145) for how to convert heatmaps into coordinates and confidence. In short, the coordinates and values of the maximum points the each heatmap is returned.

**Reply by ggggg111** [reply]
Thank you for the fast response!

I've been experimenting with what you just said, and gonna check [get_heatmap_maximum](https://github.com/open-mmlab/mmpose/blob/6d10b2ec81da7e252016b3154c7fdb46c403ecd8/mmpose/codecs/utils/post_processing.py#L145) in more detail. The problem I encounter is how to handle the cases where the heatmap contains data corresponding to more than 1 person. We could get the topk(2) values, but it will be incorrect if, e.g. the second maximum value is right next to the maximum value pixel corresponding to the same person.

**Reply by ggggg111** [reply]
@Ben-Louis Nevermind, I forgot to use a detector to detect the persons from the input video first. I just gave the input video directly to the pose estimator.
